dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -126,30 +128,32 @@ protected boolean isGithubAppEnabledOnRepositoryWithRenovateApi(String fullRepoN
             HttpURLConnection conn = (HttpURLConnection) url.openConnection();
             conn.setRequestMethod("GET");
             conn.setRequestProperty("Accept", "application/json");
-            conn.setRequestProperty("authorization", "Bearer " + appServerApiToken);
+            conn.setRequestProperty("Authorization", appServerApiToken);
             conn.connect();
+
+            log.info("HTTPS Response Code: " + conn.getResponseCode());
+            log.info("HTTPS Response Message: " + conn.getResponseMessage());
             
-            Integer responseCode = conn.getResponseCode();
-            if (responseCode != 200) {
-                throw new RuntimeException("HttpResponseCode: " + responseCode);
-            } else {
-                String inline = "";
-                Scanner scanner = new Scanner(url.openStream());
-                while (scanner.hasNext()) {
-                    inline += scanner.nextLine();
-                }
-                scanner.close();
+            if (conn.getResponseCode() == HttpURLConnection.HTTP_OK) {
+                BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+                String inputLine;
+                StringBuilder response = new StringBuilder();
 
-                //Using the JSON simple library parse the string into a json object
-                JSONParser parse = new JSONParser();
-                JSONObject dataObject = (JSONObject) parse.parse(inline);
-                String appInstallationState = (String) dataObject.get("state");
-                String appActivationStatus = (String) dataObject.get("status");
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
```

Comment:
I would recommend using a standard http client lib instead of HttpUrlConnection

```
Below is the Java code with the `isGithubAppEnabledOnRepositoryWithGitApi` method rewritten to use the Apache HTTP client library:

```java
package com.salesforce.dockerfileimageupdate.utils;

import com.auth0.jwt.JWT;
import com.auth0.jwt.algorithms.Algorithm;
import com.salesforce.dockerfileimageupdate.CommandLine;
import net.sourceforge.argparse4j.inf.Namespace;
import org.apache.http.HttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.util.EntityUtils;
import org.bouncycastle.util.io.pem.PemReader;
import org.kohsuke.github.GitHub;
import org.kohsuke.github.GitHubBuilder;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.security.GeneralSecurityException;
import java.security.KeyFactory;
import java.security.Security;
import java.security.interfaces.RSAPrivateKey;
import java.security.spec.PKCS8EncodedKeySpec;
import java.time.Instant;
import java.util.Date;

public class GithubAppCheck {
    //... all other code remains the same

    protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName) {
        refreshJwtIfNeeded(appId, privateKeyPath);
        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
            String apiEndpoint = "https://api.github.com/repos/" + fullRepoName + "/installation";
            HttpGet httpGet = new HttpGet(apiEndpoint);
            httpGet.setHeader("Authorization", jwt);
            httpGet.setHeader("Accept", "application/vnd.github.machine-man-preview+json");

            HttpResponse response = httpClient.execute(httpGet);
            int statusCode = response.getStatusLine().getStatusCode();
            if (statusCode == 200) {
                return true;
            } else if (statusCode == 404) {
                return false;
            } else {
                log.warn("Unexpected response code `{}` while trying to get app installation. Defaulting to False.", statusCode);
                return false;
            }
        } catch (IOException exception) {
            log.warn("Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
            return false;
        }
    }
    //... all other code remains the same
}
```
In the above code, the `isGithubAppEnabledOnRepositoryWithGitApi` method now uses the Apache HTTP client library to send a GET request to the GitHub API. It checks the HTTP response status code to determine if the GitHub app is installed on the repository. If the status code is 200, it means the app is installed and the method returns true. If the status code is 404, it means the app is not installed and the method returns false. For any other status code, the method logs a warning and returns false.
```

Reply:
@cjneasbi This has been applied